### PR TITLE
refactor(function): Ensure Comment Function (#4996)

### PR DESCRIPTION
**Pull request by souravdasslg**
1. Create interface EnsureCommentConfig
2. Add config to the interface for Bitbucket
3. Refactor functions
4. Update tests

#4996

### CODE REVIEW

File: lib/platform/bitbucket/index.ts [hunk 1]
```diff
@@ -653,13 +654,18 @@ export /* istanbul ignore next */ function deleteLabel(): never {
...
-): Promise<boolean> {
+export function ensureComment({
+  number: prNo,
+  subject: topic,
+  content,
+}: EnsureCommentConfig): Promise<boolean> {
   // https://developer.atlassian.com/bitbucket/api/2/reference/search?q=pullrequest+comment
-  return comments.ensureComment(config, prNo, topic, sanitize(content));
+  return comments.ensureComment({
+    config,
```
Comment (rarkins) [inline lib/platform/bitbucket/index.ts]: Surely this doesn't work currently? Cc @ViceIce

Comment (rarkins) [reply lib/platform/bitbucket/index.ts]: Sorry never mind, I see that it's `comments.ensureComment`

**Comment by rarkins** [comment]
@JamieMagee @ViceIce I'm wondering if this is an actions cache problem.

image

Works locally for me and on CircleCI.

**Comment by ViceIce** [comment]
Yes, i think so. maybe we have to disable `node_modules` caching on github actions

**Review by viceice — APPROVED** [review]

**Pull request merged**